File: frontend/src/services/deptAnnouncement.ts
import api from './api'

export interface DeptAnnouncement {
  id: number
  title: string
  note: string
  filename: string
  file_size: number
  uploaded_by: string
  uploaded_at: string
  status: '待审核' | '已发布' | '已驳回'
  reviewed_by: string
  reviewed_at: string
  reject_reason: string
}

export const listDeptAnnouncements = () =>
  api.get<{ ok: boolean; data: DeptAnnouncement[]; is_reviewer: boolean; can_upload: boolean }>(
    '/dept-announcements')

export const createDeptAnnouncement = (title: string, note: string, file?: File) => {
  const fd = new FormData()
  fd.append('title', title)
  fd.append('note', note)
  if (file) fd.append('file', file)
  return api.post<{ ok: boolean; message: string }>('/dept-announcements', fd,
    { headers: { 'Content-Type': 'multipart/form-data' } })
}

export const reviewDeptAnnouncement = (id: number, action: 'approve' | 'reject', reason = '') =>
  api.post<{ ok: boolean; message: string }>(`/dept-announcements/${id}/review`, { action, reason })

export const deleteDeptAnnouncement = (id: number) =>
  api.delete<{ ok: boolean }>(`/dept-announcements/${id}`)

export const deptAnnouncementDownloadUrl = (id: number) =>
  `/api/dept-announcements/${id}/download`

File: backend/models/dept_announcement.py
from models import db


class DeptAnnouncement(db.Model):
    """采购部公告和相关文件（分流页展示）。

    经办人上传 → 陈梦霞（分发岗）审核 → 发布后全员可见可下载。
    """
    __tablename__ = "dept_announcements"
    id          = db.Column(db.Integer, primary_key=True)
    title       = db.Column(db.String(200), default="")
    note        = db.Column(db.Text, default="")            # 说明（选填）
    filename    = db.Column(db.String(300), default="")     # 原始文件名（可空=纯公告）
    saved_name  = db.Column(db.String(100), default="")
    file_size   = db.Column(db.Integer, default=0)
    uploaded_by = db.Column(db.String(50), default="")
    uploaded_at = db.Column(db.String(30), default="")
    status      = db.Column(db.String(10), default="待审核")  # 待审核|已发布|已驳回
    reviewed_by   = db.Column(db.String(50), default="")
    reviewed_at   = db.Column(db.String(30), default="")
    reject_reason = db.Column(db.String(300), default="")

    def to_dict(self):
        return {c.name: getattr(self, c.name) for c in self.__table__.columns}
